api/controllers/deploy.go

```diff
@@ -54,30 +48,38 @@ func (d *DeployController) CreateDeploy(c *fireball.Context) (fireball.Response,
 		return nil, errors.New(errors.InvalidRequest, err)
 	}
 
-	return createJob(d.TagStore, d.JobStore, models.CreateDeployJob, req)
+	deployID, err := d.DeployProvider.Create(req)
+	if err != nil {
+		return nil, err
+	}
+
+	return fireball.NewJSONResponse(200, models.CreateEntityResponse{EntityID: deployID})
 }
 
-func (d *DeployController) DeleteDeploy(c *fireball.Context) (fireball.Response, error) {
-	id := c.PathVariables["id"]
-	return createJob(d.TagStore, d.JobStore, models.DeleteDeployJob, id)
+func (d *DeployController) deleteDeploy(c *fireball.Context) (fireball.Response, error) {
+	deployID := c.PathVariables["id"]
+	if err := d.DeployProvider.Delete(deployID); err != nil {
+		return nil, err
+	}
+
+	return fireball.NewJSONResponse(200, nil)
 }
 
-func (d *DeployController) GetDeploy(c *fireball.Context) (fireball.Response, error) {
-	id := c.PathVariables["id"]
-	model, err := d.DeployProvider.Read(id)
+func (d *DeployController) listDeploys(c *fireball.Context) (fireball.Response, error) {
+	deploys, err := d.DeployProvider.List()
 	if err != nil {
 		return nil, err
 	}
 
-	return fireball.NewJSONResponse(200, model)
+	return fireball.NewJSONResponse(200, deploys)
 }
 
-func (d *DeployController) ListDeploys(c *fireball.Context) (fireball.Response, error) {
-	summaries, err := d.DeployProvider.List()
+func (d *DeployController) readDeploy(c *fireball.Context) (fireball.Response, error) {
```

Comment:
Curious why you opted for `read` versus `get`

Reply:
with the exception of the name of the cli command, everywhere else in our codebase uses `read` instead of `get`.